runtime/src/info.rs

```diff
@@ -80,3 +81,42 @@ impl IDevIdCertCmd {
         }
     }
 }
+
+pub struct PopulateIDevIdCertCmd;
+impl PopulateIDevIdCertCmd {
+    pub(crate) fn execute(drivers: &mut Drivers, cmd_args: &[u8]) -> CaliptraResult<MailboxResp> {
+        if let Some(cmd) = PopulateIdevCertReq::read_from(cmd_args) {
+            let cert_size = cmd.cert_size as usize;
+            if cert_size > cmd.cert.len() {
+                return Err(CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS);
+            }
+
+            let flags = drivers.persistent_data.get().manifest1.header.flags;
+            // PL1 cannot call this mailbox command
+            if flags & PL0_PAUSER_FLAG == 0 {
+                return Err(CaliptraError::RUNTIME_INCORRECT_PAUSER_PRIVILEGE_LEVEL);
+            }
+
+            // Insert IDevId certificate at the start of the certificate chain.
+            for i in 0..cert_size {
+                if i > drivers.cert_chain.len() {
+                    return Err(CaliptraError::RUNTIME_IDEV_CERT_POPULATION_FAILED);
+                }
+
+                drivers
+                    .cert_chain
+                    .try_insert(
```

Comment:
I believe every `try_insert` call will result in copying every element in the ArrayVec.

Another option for implementing this would to do something like:

```rust
let tmp_chain: ArrayVec::<u8, MAX_CERT_CHAIN_SIZE>::new();
tmp_chain.try_extend_from_slice(&cmd.cert.get(..cmd.cert.len());
tmp_chain.try_extend_from_slice(drivers.cert_chain.as_slice());
drivers.cert_chain = tmp_chain;
```

Ends up being less code and also more efficient.

Reply:
Thanks, this is much nicer. Done.